kernel/src/syscall/stat.rs

```diff
@@ -15,9 +15,13 @@ use crate::{
 pub fn sys_fstat(fd: FileDesc, stat_buf_ptr: Vaddr, ctx: &Context) -> Result<SyscallReturn> {
     debug!("fd = {}, stat_buf_addr = 0x{:x}", fd, stat_buf_ptr);
 
-    let file_table = ctx.process.file_table().lock();
-    let file = file_table.get_file(fd)?;
-    let stat = Stat::from(file.metadata());
+    let file_metadata = {
+        let file_table = ctx.process.file_table().lock();
+        let file = file_table.get_file(fd)?;
+        file.metadata()
```

Comment:
Move `file.metadata()` out of this code block to further reduce the scope of holding the spinlock.

Reply:
It's a tradeoff here, moving `file.metadata()` out requires a clone on `file: Arc<dyn FileLike>`, which could negatively impact `lmbench-stat`.

Reply:
In theory, getting the metadata of a file may also involve I/O or sleeping logic. So it definitely should not be done while holding a spinlock.